pkg/provider/provider.go

```diff
@@ -96,6 +96,15 @@ func (e FlagProvider) resolveFlag(ctx context.Context, flag string, defaultValue
 	return processResolvedFlag(resolvedFlag, defaultValue, expectedKind, propertyPath)
 }
 
+func processTargetingKey(evalCtx openfeature.FlattenedContext) openfeature.FlattenedContext {
+	newEvalContext := openfeature.FlattenedContext{}
+	newEvalContext = evalCtx
+	if targetingKey, exists := evalCtx["targetingKey"]; exists {
+		newEvalContext["targeting_key"] = targetingKey
+	}
+	return newEvalContext
+}
```

Comment:
Can't say much about the Go specifics but looks to do the right thing.
Would love to see a unit test for this!